src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -1240,4 +1254,29 @@ namespace smt::noodler {
         }
     }
 
+    /**
+     * @brief Check if the instance is clearly unsatisfiable.
+     * 
+     * @return True --> unsat for sure.
+     */
+    bool FormulaPreprocessor::contains_unsat_predicates() const {
```

Comment:
Maybe `contains_unsat_eqs_or_diseqs` or something similar would be a better name.

Reply:
Fixed